### ISSUE TITLE
Gym Archive Fails -- Can't find provisioning profile downloaded by match

### ISSUE THREAD

**Issue by IndianAg0711** [fastlane/fastlane#8644]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
When gym runs I get this error:
![image](https://cloud.githubusercontent.com/assets/6537058/24304613/fbcf74ee-1087-11e7-90c1-fd573fe1710b.png)

And in the logs I get this:

```
No profile matching 'match AppStore xxx' found:  Xcode couldn't find a profile matching 'match AppStore xxx'. Install the profile (by dragging and dropping it onto Xcode's dock item) or select a different one in the General tab of the target editor.
Code signing is required for product type 'Application' in SDK 'iOS 10.2'
```

Although I can clearly see the profiles installed through xcode. I've also walked through the code signing xcode guide to make sure that I explicitly set the profiles I want to use versus letting xcode manage automatically.

The only thing running in my fastfile:
``` 
lane :beta do
    match(type: "appstore", git_url: "xxx")
    gym(scheme: "xxx", project: "./ios/xxx.xcodeproj")
    pilot
end
```
### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.11.6                                     |
| Ruby                        | 2.2.3                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.10.1 (Apple Git-78)           |
| Installation Source         | ~/.rbenv/versions/2.2.3/bin/fastlane        |
| Host                        | Mac OS X 10.11.6 (15G1217)                  |
| Ruby Lib Dir                | ~/.rbenv/versions/2.2.3/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2k  26 Jan 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.2.1                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.23.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."


  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    match(type: "appstore", git_url: "xxx")
    gym(scheme: "xxx", project: "xxx")
    pilot
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    # match(type: "appstore")
    # snapshot
    gym(scheme: "xxx") # Build your app - more options available
    deliver(force: true)
    # frameit
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    #
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "xxx" # The bundle identifier of your app
apple_id "xxx" # Your Apple email address

team_id "xxx"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.23.0  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| bundler                   | 1.14.6       |
| CFPropertyList            | 2.3.5        |
| i18n                      | 0.8.1        |
| minitest                  | 5.10.1       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.2        |
| activesupport             | 4.2.8        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.0        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170223 |
| dotenv                    | 2.2.0        |
| excon                     | 0.55.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.11.0       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| jwt                       | 1.5.6        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.0        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| google-api-client         | 0.9.28       |
| json                      | 2.0.3        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.2.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| tty-screen                | 0.5.0        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| xcpretty-travis-formatter | 0.0.4        |
</details>

*generated on:* **2017-03-24**
</details>

**Comment by IndianAg0711** [comment]
I've used the troubleshooting guide to no avail. Anyone have any guesses at what I might be doing wrong?

**Comment by mgrebenets** [comment]
@IndianAg0711 
One simple check to go through - does it work when you run `Archive` action in Xcode?
Or directly run `xcodebuild` command?

As a matter of fact, you can look through the `fastlane` logs and see the actual shell script it runs.
It usually looks like `set -o pipefail && xcodebuild ...`

This is the command you can try to run directly your self, you can even throw away some custom parts of it, but you will be able to reproduce the issue in the end.

When it comes to building a project with `gym` it will end up as `xcodebuild` command all the same.
If that command fails as part of `fastlane`, it fails directly as well.

**Comment by IndianAg0711** [comment]
@mgrebenets 
Thanks for the help! When I tried to archive the app in xcode initially I got back an error because the required app icon I was using was the wrong pixel size. Fixing this allowed me to manually archive in xcode, but for some reason I still get the original error when I try deploying via the fastlane cli. The end goal is to get bitrise doing this automatically, so I might just try pushing over there and seeing if it has better luck than my local machine.
